src/overlays/actors/ovl_Obj_Lift/z_obj_lift.c

```diff
@@ -28,46 +38,207 @@ const ActorInit Obj_Lift_InitVars = {
+
+void func_8093D3C0(ObjLift* this, PlayState* play) {
+    Vec3f pos;
+    Vec3f vel;
+    Vec3f* actorPos = &this->dyna.actor.world.pos;
+    s32 i;
+    s32 rnd;
+    s32 pad;
+
+    for (i = 0; i < ARRAY_COUNT(D_8093DD60); i++) {
+        pos.x = (D_8093DD60[i].x * this->dyna.actor.scale.x) + actorPos->x;
+        pos.y = actorPos->y;
+        pos.z = (D_8093DD60[i].z * this->dyna.actor.scale.z) + actorPos->z;
+
+        vel.x = D_8093DD60[i].x * this->dyna.actor.scale.x * 0.8f;
+        vel.y = (Rand_ZeroOne() * 10.0f) + 6.0f;
+        vel.z = D_8093DD60[i].z * this->dyna.actor.scale.z * 0.8f;
+
+        if ((s32)Rand_Next() > 0) {
+            rnd = 0x40;
+        } else {
+            rnd = 0x20;
+        }
+
+        EffectSsKakera_Spawn(play, &pos, &vel, actorPos, -0x100, rnd, 15, 15, 0,
+                             ((Rand_ZeroOne() * 50.0f) + 50.0f) * this->dyna.actor.scale.x, 0, 32, 50, -1,
+                             OBJECT_D_LIFT, gDampeGraveBrownElevatorDL);
+    }
+    if (OBJLIFT_GET_1(&this->dyna.actor) == 0) {
+        func_800BBFB0(play, &this->dyna.actor.world.pos, 120.0f, 12, 120, 100, 1);
+    } else if (OBJLIFT_GET_1(&this->dyna.actor) == 1) {
+        func_800BBFB0(play, &this->dyna.actor.world.pos, 60.0f, 8, 60, 100, 1);
+    }
+}
+
+void ObjLift_Init(Actor* thisx, PlayState* play) {
+    f32 temp_fv0;
+    ObjLift* this = THIS;
+
+    Actor_ProcessInitChain(&this->dyna.actor, sInitChain);
+    this->dyna.actor.scale.x = this->dyna.actor.scale.z = D_8093DD98[OBJLIFT_GET_1(&this->dyna.actor)];
+    this->dyna.actor.scale.y = 5.0f / 90.0f;
+    this->dyna.actor.shape.rot.z = 0;
+    this->unk_178 = this->dyna.actor.home.rot.z;
+    this->dyna.actor.home.rot.z = this->dyna.actor.world.rot.z = this->dyna.actor.shape.rot.z;
+    DynaPolyActor_Init(&this->dyna, 1);
+    if ((this->unk_178 <= 0) && (Flags_GetSwitch(play, OBJLIFT_GET_7F(&this->dyna.actor)))) {
+        Actor_MarkForDeath(&this->dyna.actor);
+    } else {
+        DynaPolyActor_LoadMesh(play, &this->dyna, &gDampeGraveBrownElevatorCol);
+        this->unk_160 = Rand_Next() >> 0x10;
+        this->unk_162 = Rand_Next() >> 0x10;
+        this->unk_164 = Rand_Next() >> 0x10;
+        func_8093D760(this);
+    }
+}
+
+void ObjLift_Destroy(Actor* thisx, PlayState* play) {
+    ObjLift* this = THIS;
+
+    DynaPoly_DeleteBgActor(play, &play->colCtx.dyna, this->dyna.bgId);
+}
+
+void func_8093D760(ObjLift* this) {
+    this->timer = D_8093DD50[OBJLIFT_GET_7(&this->dyna.actor)];
+    this->actionFunc = func_8093D7A0;
+    this->dyna.actor.draw = ObjLift_Draw;
+}
+
+void func_8093D7A0(ObjLift* this, PlayState* play) {
+    s32 pad;
+    s16 quake;
+
+    if (DynaPolyActor_IsInRidingMovingState(&this->dyna)) {
+        if (this->timer <= 0) {
+            if (OBJLIFT_GET_7(&this->dyna.actor) == 7) {
+                func_8093D9C0(this);
+            } else {
+                quake = Quake_Add(play->cameraPtrs[play->activeCamera], 1);
```

Comment:
```suggestion
                quake = Quake_Add(GET_ACTIVE_CAM(play), 1);
```
There's a macro you can be using here